dace/transformation/interstate/loop_to_map.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2019-2020 ETH Zurich and the DaCe authors. All rights reserved.
+""" Loop to map transformation """
+
+import copy
+import sympy as sp
+import networkx as nx
+from typing import List, Optional, Tuple
+
+from dace import dtypes, memlet, nodes, registry, sdfg as sd, symbolic, subsets
+from dace.properties import Property, make_properties, CodeBlock
+from dace.sdfg import graph as gr, nodes
+from dace.sdfg import utils as sdutil
+from dace.frontend.python.astutils import ASTFindReplace
+from dace.transformation.interstate.loop_detection import (DetectLoop,
+                                                           find_for_loop)
+import dace.transformation.helpers as helpers
+
+
+@registry.autoregister
+class LoopToMap(DetectLoop):
+    """Convert a control flow loop into a dataflow map. Currently only supports
+       the simple case where there is no overlap between inputs and outputs in
+       the body of the loop, and where the loop body only consists of a single
+       state.
+    """
+
+    @staticmethod
+    def can_be_applied(graph, candidate, expr_index, sdfg, strict=False):
+        # Is this even a loop
+        if not DetectLoop.can_be_applied(graph, candidate, expr_index, sdfg,
+                                         strict):
+            return False
+
+        guard = graph.node(candidate[DetectLoop._loop_guard])
+        begin = graph.node(candidate[DetectLoop._loop_begin])
+
+        # Guard state should contain any dataflow
+        if len(guard.nodes()) != 0:
+            return False
+
+        # Only support loops with a single-state body
+        begin_outedges = graph.out_edges(begin)
+        if len(begin_outedges) != 1 or begin_outedges[0].dst != guard:
+            return False
+
+        # If loop cannot be detected, fail
+        _, rng = find_for_loop(graph, guard, begin)
```

Comment:
this will not return a 2-tuple on failure, but None. This check will raise an exception